File: ruby/testcode/ssrf_017.rb
require_relative 'shared'

module Curb
  class Easy
    def self.perform(url); new(url); end
    def initialize(url); @url = url; end
    def body_str; @url; end
  end
end

# vuln-code-snippet start ruby_ssrf_curb
def fetch_curb(req)
  url = req.param('url')
  response = Curb::Easy.perform(url) # vuln-code-snippet vuln-line ruby_ssrf_curb
  BenchmarkResponse.ok(response.body_str)
end
# vuln-code-snippet end ruby_ssrf_curb

File: php/testcode/ssrf_017.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_ssrf_curl_open_url
function ssrf017(BenchmarkRequest $req): BenchmarkResponse {
    $url = $req->param('url');
    $ch = curl_init();
    curl_setopt($ch, CURLOPT_URL, $url); // vuln-code-snippet vuln-line php_ssrf_curl_open_url
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    $result = curl_exec($ch);
    curl_close($ch);
    return BenchmarkResponse::ok($result);
}
// vuln-code-snippet end php_ssrf_curl_open_url
